certificate-service/clinicalprocess-certificate-v4/src/main/java/se/inera/intyg/certificateservice/infrastructure/clinicalprocesscertificatev4/certificate/XmlGeneratorDateRange.java

```diff
@@ -0,0 +1,69 @@
+package se.inera.intyg.certificateservice.infrastructure.clinicalprocesscertificatev4.certificate;
+
+import java.time.LocalDate;
+import java.util.Collections;
+import java.util.List;
+import javax.xml.datatype.DatatypeFactory;
+import javax.xml.datatype.XMLGregorianCalendar;
+import org.springframework.stereotype.Component;
+import se.inera.intyg.certificateservice.domain.certificate.model.ElementData;
+import se.inera.intyg.certificateservice.domain.certificate.model.ElementValue;
+import se.inera.intyg.certificateservice.domain.certificate.model.ElementValueDateRange;
+import se.inera.intyg.certificateservice.domain.certificatemodel.model.ElementSpecification;
+import se.riv.clinicalprocess.healthcond.certificate.types.v3.DatePeriodType;
+import se.riv.clinicalprocess.healthcond.certificate.types.v3.ObjectFactory;
+import se.riv.clinicalprocess.healthcond.certificate.v3.Svar;
+import se.riv.clinicalprocess.healthcond.certificate.v3.Svar.Delsvar;
+
+@Component
+public class XmlGeneratorDateRange implements XmlGeneratorElementData {
+
+  @Override
+  public Class<? extends ElementValue> supports() {
+    return ElementValueDateRange.class;
+  }
+
+  public List<Svar> generate(ElementData data, ElementSpecification specification) {
+    if (!(data.value() instanceof ElementValueDateRange dateRangeValue)) {
+      return Collections.emptyList();
+    }
+
+    if (dateRangeValue.toDate() == null && dateRangeValue.fromDate() == null) {
+      return Collections.emptyList();
+    }
+
+    final var objectFactory = new ObjectFactory();
+
+    final var dateRangeAnswer = new Svar();
+    dateRangeAnswer.setId(
```

Comment:
We should always be able to use the id from data, so I think we can update this to

```java
  final var dateRangeAnswer = new Svar();
  dateRangeAnswer.setId(data.id().id());
```